infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -146,85 +143,60 @@ public synchronized void delete(@Nonnull String key,
                     }
                 })
                 .maximumSize(dsCacheSize)
-                .build(key -> {
-                    try {
-                        Path path = Paths.get(logDir + File.separator + key + EXTENSION);
-                        if (Files.notExists(path)) {
-                            return null;
-                        }
-                        byte[] bytes = Files.readAllBytes(path);
-                        ByteBuffer buf = ByteBuffer.wrap(bytes);
-                        int checksum = buf.getInt();
-                        byte[] strBytes = Arrays.copyOfRange(bytes, 4, bytes.length);
-                        if (checksum != getChecksum(strBytes)) {
-                            throw new DataCorruptionException();
-                        }
-                        return new String(strBytes);
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
-                    }
-                });
+                .build();
 
         return cache;
     }
 
     @Override
     public synchronized  <T> void put(Class<T> tclass, String prefix, String key, T value) {
-        cache.put(getKey(prefix, key), JsonUtils.parser.toJson(value, tclass));
+        cache.put(getKey(prefix, key), value);
     }
 
-    @Override
-    public synchronized  <T> T get(Class<T> tclass, String prefix, String key) {
-        String json = cache.get(getKey(prefix, key));
-        return getObject(json, tclass);
-    }
+    private <T> T load(Class<T> tClass, String key) {
+        try {
+            Path path = Paths.get(logDir + File.separator + key + EXTENSION);
+            if (Files.notExists(path)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3725)